src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,44 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
-}
+  // write your code here0
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 
+  hide() {
+    this.hidden = true;
+  }
+}
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  bite(enemy) {
+    if (enemy instanceof Herbivore && enemy.hidden === false) {
+      enemy.health -= 50;
+
+      if (enemy.health <= 0) {
+        Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
[CODE PERFOMANCE] You're currently filtering the `Animal.alive` array every time an animal's health reaches 0. This might not be the most efficient approach, especially if the `Animal.alive` array is very large. Instead, you could use the `Array.prototype.splice()` method to remove the dead animal from the array. This would avoid the need to iterate over the entire array.